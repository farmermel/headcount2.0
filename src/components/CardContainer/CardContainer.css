.card-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

@media screen and (max-width: 720px) {
  .card-container {
    font-size: 12px;
  }
}