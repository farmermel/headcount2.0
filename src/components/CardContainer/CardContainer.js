import React from 'react';
import Card from '../Card/Card';
import PropTypes from 'prop-types';
import './CardContainer.css';

const CardContainer = ({ districts, compare, comparison }) => {
  const allDistricts = districts.map(district => {
    return <Card {...district} key={district.location} compare={compare} />;
  });

  return <div className="card-container">{allDistricts}</div>;
};

CardContainer.propTypes = {
  districts: PropTypes.arrayOf(PropTypes.object)
};

export default CardContainer;
