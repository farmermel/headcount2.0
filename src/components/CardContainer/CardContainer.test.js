import React from 'react';
import ReactDOM from 'react-dom';
import CardContainer from './CardContainer';
import { shallow, mount } from 'enzyme';

describe('CardContainer', () => {
  let wrapper;
  const mockDistricts = [{cow: 'moo'}]

  it('should match the snapshot', () => {
    wrapper = shallow(<CardContainer districts={mockDistricts} />);
    expect(wrapper).toMatchSnapshot();
  })
})
