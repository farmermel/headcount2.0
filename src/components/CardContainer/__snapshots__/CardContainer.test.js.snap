// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    compare={[Function]}
    comparison={
        Object {
            "cow": Object {
              "otherCow": 7,
            },
          }
    }
    districts={
        Array [
            Object {
              "cow": "moo",
              "location": "farm",
              "selected": false,
            },
            Object {
              "location": "mud",
              "pig": "oink",
              "selected": false,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          compare={[Function]}
          cow="moo"
          location="farm"
          selected={false}
/>,
        <Card
          compare={[Function]}
          location="mud"
          pig="oink"
          selected={false}
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "farm",
        "nodeType": "function",
        "props": Object {
          "compare": [Function],
          "cow": "moo",
          "location": "farm",
          "selected": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "mud",
        "nodeType": "function",
        "props": Object {
          "compare": [Function],
          "location": "mud",
          "pig": "oink",
          "selected": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            compare={[Function]}
            cow="moo"
            location="farm"
            selected={false}
/>,
          <Card
            compare={[Function]}
            location="mud"
            pig="oink"
            selected={false}
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "farm",
          "nodeType": "function",
          "props": Object {
            "compare": [Function],
            "cow": "moo",
            "location": "farm",
            "selected": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "mud",
          "nodeType": "function",
          "props": Object {
            "compare": [Function],
            "location": "mud",
            "pig": "oink",
            "selected": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
